### PR TITLE
docs(conditions): add examples for conditions

### PR DESCRIPTION
**Pull request by wjgogogo**
### 👀 PR includes

🎨 Enhance

- [x] refactor interval condition judgement
- [x] fix electron version to adapt mac M1

🔧 Chore

- [x] add markdownlint for ci and pre-commit
- [x] add examples for conditons
- [x] complete condition api docs 
- [x] condition manual

### 🖼️ Screenshot
examples:
image
image

api:
image

### CODE REVIEW

**Comment by xingwanying** [comment]
image

配色有些死亡，参考 antv 色板
https://antv.vision/zh/docs/specification/language/palette

File: s2-site/docs/api/general/conditions.zh.md [hunk 1]
```diff
@@ -0,0 +1,124 @@
+---
+title: 字段标记
+order: 2
+redirect_from:
```
Comment (xingwanying) [inline s2-site/docs/api/general/conditions.zh.md:4]: redirect_from 可以删掉

File: s2-site/docs/api/general/conditions.en.md [hunk 2]
```diff
@@ -0,0 +1,7 @@
+---
+title: Conditions
+order: 2
+redirect_from:
+  - /en/docs/api
```
Comment (xingwanying) [inline s2-site/docs/api/general/conditions.en.md:5]: redirect_from 可以删掉

File: s2-site/docs/api/general/conditions.zh.md [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+  - /zh/docs/api
+---
+
+## Conditions
+
+类型：`object`，**必选**，默认值：`{}`
```
Comment (xingwanying) [inline s2-site/docs/api/general/conditions.zh.md:10]: 这部分的参数列表可以放 `S2/s2-site/docs/common` 里，然后 `options` 和这里都能直接引用，公共部分都放 `common` 吧

**Review by lijinke666 — APPROVED** [review]

**Pull request merged**